packages/client-core/src/social/services/PartyService.ts

```diff
@@ -10,104 +9,99 @@ import { Party } from '@xrengine/common/src/interfaces/Party'
 
 // State
-const state = createState({
-  party: {} as Party,
-  updateNeeded: true
+const PartyState = defineState({
+  name: 'PartyState',
+  initial: () => ({
+    party: {} as Party,
+    updateNeeded: true
+  })
 })
 
-store.receptors.push((action: PartyActionType): any => {
-  let newValues, updateMap, partyUser, updateMapPartyUsers
-
-  state.batch((s) => {
-    switch (action.type) {
-      case 'LOADED_PARTY':
+export const PartyServiceReceptor = (action) => {
+  getState(PartyState).batch((s) => {
+    // let newValues, updateMap, partyUser, updateMapPartyUsers
+    matches(action)
+      .when(PartyAction.loadedPartyAction.matches, (action) => {
         return s.merge({ party: action.party, updateNeeded: false })
-      case 'CREATED_PARTY':
+      })
+      .when(PartyAction.createdPartyAction.matches, () => {
         return s.updateNeeded.set(true)
-      case 'REMOVED_PARTY':
-        updateMap = new Map()
+      })
+      .when(PartyAction.removedPartyAction.matches, () => {
         return s.merge({ party: {}, updateNeeded: true })
-      case 'INVITED_PARTY_USER':
+      })
+      .when(PartyAction.invitedPartyUserAction.matches, () => {
         return s.updateNeeded.set(true)
-      case 'CREATED_PARTY_USER':
-        newValues = action
-        partyUser = newValues.partyUser
-        updateMap = _.cloneDeep(s.party.value)
+      })
+      .when(PartyAction.createdPartyUserAction.matches, (action) => {
+        const updateMap = _.cloneDeep(s.party.value)
         if (updateMap != null) {
-          updateMapPartyUsers = updateMap.partyUsers
-          updateMapPartyUsers = Array.isArray(updateMapPartyUsers)
-            ? updateMapPartyUsers.find((pUser) => {
-                return pUser != null && pUser.id === partyUser.id
+          updateMap.partyUsers = Array.isArray(updateMap.partyUsers)
+            ? updateMap.partyUsers.find((pUser) => {
+                return pUser != null && pUser.id === action.partyUser.id
               }) == null
-              ? updateMapPartyUsers.concat([partyUser])
+              ? updateMap.partyUsers.concat([action.partyUser])
               : updateMap.partyUsers.map((pUser) => {
-                  return pUser != null && pUser.id === partyUser.id ? partyUser : pUser
+                  return pUser != null && pUser.id === action.partyUser.id ? action.partyUser : pUser
                 })
-            : [partyUser]
-          updateMap.partyUsers = updateMapPartyUsers
+            : [action.partyUser]
         }
         return s.merge({ party: updateMap, updateNeeded: true })
-
-      case 'PATCHED_PARTY_USER':
-        newValues = action
-        partyUser = newValues.partyUser
-        logger.info({ partyUser }, 'Patched partyUser.')
-        updateMap = _.cloneDeep(s.party.value)
+      })
+      .when(PartyAction.patchedPartyUserAction.matches, (action) => {
+        logger.info({ partyUser: action.partyUser }, 'Patched partyUser.')
```

Comment:
please remove all logs like this, they are unnecessary now as all actions are automatically logged

Reply:
There are some logs from the socket, I think I can keep them.